Source/Scene/Scene.js

```diff
@@ -1851,12 +2012,40 @@ define([
     function updatePrimitives(scene) {
         var frameState = scene._frameState;
 
+        scene._groundPrimitives.update(frameState);
+        scene._primitives.update(frameState);
+
+        var shadowMaps = frameState.shadowMaps;
+        var length = shadowMaps.length;
+        for (var i = 0; i < length; ++i) {
+            var shadowMap = shadowMaps[i];
+            shadowMap.update(frameState);
+
+            if (!shadowMap.outOfView && defined(scene._globe) && scene._globe.castShadows) {
+                var sceneCamera = frameState.camera;
```

Comment:
Instead of save/set/restore, is there a cleaner way to do this?  Perhaps create a separate frame state for the shadow map?